### ISSUE TITLE
Accessibility issues flagged by Jeff and in reports

### ISSUE THREAD

**Issue by tomay** [GreenInfo-Network/seattle-building-dashboard#152]
First, a list of things that Jeff has injected into live code that we need to replicate

* [x] Leaflet map tile PNGs need alt tags to avoid being flagged in automatic tests. This is currently done in a timeout in the router (but likely only applies to tiles in the view, not all possible dynamically loaded tiles). **RESOLUTION**: Added `aria-label="hidden"` to the entire map. Ultimately, the map should have a dynamic caption to describe what the map contains, but for now, screen readers will skip the map. 
* [x] Address search icon is given an alt of 'search'. But really, this is a decorative icon, and can probably have empty alt, or use an aria-hidden tag. **RESOLUTION**: I added both `alt=''` and `aria-hidden='true'`
* [x] Add `lang="en"` to root `<html>` tag in index.html
* [x] Also in index.html, in meta content, sets `maximum-scale` to 2.0. We should look this up. I think we remove maximum scale **RESOLUTION**: I removed the `maximum-scale`. My read on accessibility is that you should not limit or prevent scalability
* [x] Also in index.html, adds a tabindex="0" to the div with id="map-controls-content--inner". Need to check the reasoning there. **DONE**
* [x] main.min.css: removes `outline: none` from `input:focus, select:focus, textarea:focus`. Agree with the general idea, but the implementation could be improved: 

![](https://github.com/user-attachments/assets/4d752beb-7caa-4d0f-a380-bbaa214207bf)

**DONE. Also spent some time reorganizing the search input so focus style works a bit better**

* [x] main.min.css: removes `outline: none;` from `a:active` and `a:focus`. I'm a little surprised that links need focus styles? Here's what that looks like: 

![](https://github.com/user-attachments/assets/c289fce8-03f2-45b1-b760-7d05650e4665)

**DONE**

Here is the full annotated list of changes sent by Jeff B.:
> 
> Minor Accessibility Fixes For Energy Benchmarking Map
> 
> wwwroot/energybenchmarkingmap/
> 
> index.html
> 
> screenshot_index1.png
>   Line 2 - added lang="en"
>   Line 8 - changed maximium-scale value to 2.0
> screenshot_index2.png
>   Line 73 - added tabindex="0"
> 
> /app/router.js
> 
> screenshot_app_routerJS.png
> Line 156 - Added a js hack to add a decorative alt tag to the map tilesets since Carto does not add alt text to map tiles, this does not include the tilesets that contain the building points
> 
> 
> /app/templates/map/address_search.html
> 
> Screenshot_app_templates_map_address_search.png
> Line 3 - added alt text to the search icon
> 
> 
> /app/templates/scorecard/building.html
> 
> screenshot_app_templates_scorecard_building.png
> Line 97 - Added the text Clean Buildings Performance Standard (CBPS). This was a request from the business and not related to accessibility remediation. 
> 
> /styles/main.min.css
> 
> screenshot_styles_mainmincss1.png
> Line 22 - Commented out a CSS class that was preventing focus on an element when tab is used. 
> 
> screenshot_styles_mainmincss2.png
> Line 88 - Commented out a CSS property outline="none" that was preventing focus on an element when tab is used.
> 
> screenshot_styles_mainmincss3.png
> Line 213 - Commented out a CSS class that was preventing focus on the a element when tab is used.
>

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201108652060192/1210862977153598)

**Comment by tomay** [comment]
Jeff also sent a set of screenshots from the SiteImprove Accessibility Check plugin. The screenshots are of limited value, we can't drill into the 52 instances of "Text not included in an ARIA landmark" for example. So we'll either have to run this plugin on our own or similar to get full details.

I've saved the screenshots to drive here:
https://drive.google.com/drive/folders/1wKvHkugas2p6NHlkiaXyrd4spEraEcB4?usp=drive_link

Some of these are relatively straightforward, e.g. "headings structure". 

Many others however, imply a complete redesign and rebuild, and will be out of scope, and/or not possible within this budget. 
Examples:
- "Text clipping when resized"
- Color contrast of the color ramps, and these same colors applied to text in popups and charts
- Form fields missing labels

**Comment by tomay** [comment]
Notes on other issues not flagged by SiteImprove:

## Search form at top of page
* The top bar does not directly use native form elements (input, select), and these are completely inaccessible to keyboard navigation
* None of the form elements in the top bar are correctly or clearly labelled

## Sidebar
* The sidebar cannot be reached by the keyboard
* The accordions cannot be reached or operated by keyboard

## Color contrast
* All of the charts fail color contrast. The second and third color on this ramp, for example, have a contrast ratio of 1.05 to 1, about as bad as it gets:

<img width="318" height="114" alt="Image" src="https://github.com/user-attachments/assets/c2a7de6e-7379-44b0-8716-2479d32b6356" />
 
* Colors are likely not color blind safe (simulation of how someone with 80% deuteranopia would see it)

<img width="433" height="421" alt="Image" src="https://github.com/user-attachments/assets/2c1fadc6-cc0f-42fb-a44c-2b2d392db177" />

## Comparable experience for charts

## Comparable experience for the map
